dftimewolf/lib/processors/turbinia_gcp.py

```diff
@@ -0,0 +1,131 @@
+from turbinia import config
+from turbinia import evidence
+from turbinia.pubsub import TurbiniaRequest
+from turbinia import TurbiniaException
+from turbinia import output_manager
+
+
+class TurbiniaProcessor(BaseModule):
+  """Process cloud disks with remote Turbinia instance.
+
+  Attributes:
+    disk_name: Name of the disk to process
+    project: The project containing the disk to process
+    zone: The zone containing the disk to process
+    _output_path: The path to output files
+  """
+
+  def __init__(self, state):
+    """Initialize the Turbinia artifact processor object.
+
+    Args:
+      verbose: Boolean indicating if to use verbose output
+    """
+    super(TurbiniaProcessor, self).__init__(state)
+    self.disk_name = None
+    self.project = None
+    self.zone = None
+    self._output_path = None
+
+  def setup(self, disk_name, project, zone): # pylint: disable=arguments-differ
+    """Sets up the object attributes.
+
+    Args:
+      disk_name: Name of the disk to process
+      project: The project containing the disk to process
+      zone: The zone containing the disk to process
+    """
+    if disk_name is None or project is None or zone is None:
+      self.state.add_error(
+          'disk_name, project or zone are not all specified in the recipe, '
+          'bailing out', critical=True)
+    self.disk_name = disk_name
+    self.project = project
+    self.zone = zone
+    self._output_path = tempfile.mkdtemp()
+
+  def cleanup(self):
+    pass
+
+  def process(self):
+    """Process files with Turbinia."""
+    log_file_path = os.path.join(self.output_path, 'turbinia.log')
+    self.console_out.VerboseOut('Log file: {0:s}'.format(log_file_path))
+
+    evidence_ = evidence.GoogleCloudDisk(
+        disk_name=self.disk_name, project=self.project, zone=self.zone)
+    request = TurbiniaRequest()
+    request.evidence.append(evidence_)
+
+    try:
+      config.LoadConfig()
+      region = config.TURBINIA_REGION
+      instance = config.PUBSUB_TOPIC
+      project = config.PROJECT
+      client = turbinia_client.TurbiniaClient()
+
+      self.console_out.VerboseOut(
+          'Creating Turbinia request {0:s} with Evidence {1:s}'.format(
+              request.request_id, evidence_.name))
+      client.send_request(request)
+      self.console_out.VerboseOut('Waiting for Turbinia request {0:s} to '
+                                  'complete'.format(request.request_id))
+      client.wait_for_request(
+          instance=instance, project=project, region=region,
+          request_id=request.request_id)
+      task_data = client.get_task_data(
+          instance=instance, project=project, region=region,
+          request_id=request.request_id)
+      print client.format_task_status(
+          instance=instance, project=project, region=region,
+          request_id=request.request_id, all_fields=True)
+    except TurbiniaException as e:
+      self.state.add_error(e, critical=True)
+
```

Comment:
Please `return` after adding a critical error, otherwise your code will continue executing after this. I know it's not super friendly, I added #80 to remind me to fix this.